proxy/http2/Http2ClientSession.cc

```diff
@@ -326,6 +327,9 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_WRITE_COMPLETE:
+    if (this->get_half_close_flag()) {
```

Comment:
Good point. As far as I understand, there're 2 scenarios for sending GOAWAY frame, Graceful Shutdown and Connection Error Handling. I'm sure that it is better to proceed outstanding streams as long as possible for Graceful Shutdown (maybe until active or inactive timeout). Should we do same thing for Connection Error Handling?